### ISSUE TITLE
Che 7.18.0 - Failed to run the workspace: "Error occured while waiting for Pod creation" : Websocket connection closed before Pod creation event received

### ISSUE THREAD

**Issue by jegithubpro** [eclipse/che#21037]
### Describe the bug

Launching an eclipse che 7.18.0  workspace failed suddently with the error below :

> Failed to run the workspace: "Error occured while waiting for Pod creation" : Websocket connection closed before Pod creation event received

Here after the logs of the che server.

The communication with the kubernetes API was working with the kubectl command inside the che pod.
here after the command to enable the kubectl comand from the che pod
`
apt update
apt install curl
curl -LO https://dl.k8s.io/release/v1.12.0/bin/linux/amd64/kubectl
chmod +x kubectl
./kubectl get pod -n che
NAME READY STATUS RESTARTS AGE
che-f6965b967-w8vcm 1/1 Running 4 111d
...

`
After restarting the che pod and after at least 1h the behavior come back to normal.

Can you have an explanation and a quick solution if this error occurred again ?
 

### Che version

other (please specify in additional context)

### Steps to reproduce

Simply starting a workspace.

### Expected behavior

Startup of the workspace succeed.

### Runtime

Kubernetes (vanilla), other (please specify in additional context)

### Installation method

other (please specify in additional context)

### Environment

other (please specify in additional context)

### Eclipse Che Logs

```shell
2022-01-18 09:02:01,659[nio-8080-exec-6]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 480]   - Starting workspace 'algodeveloper1/S3FS-0.0.13-JES' with id 'workspacez53xd6hwh98217ok' by user 'algodeveloper1'
2022-01-18 09:02:01,727[aceSharedPool-1]  [WARN ] [i.f.k.c.i.VersionUsageUtils 60]      - The client is using resource type 'ingresses' with unstable version 'v1beta1'
2022-01-18 09:02:29,191[/10.247.0.1/...]  [WARN ] [k.c.d.i.WatchConnectionManager 206]  - Exec Failure
java.lang.NullPointerException: null
        at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesDeployments$CreateWatcher.eventReceived(KubernetesDeployments.java:1100)
        at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesDeployments$CreateWatcher.eventReceived(KubernetesDeployments.java:1084)
        at io.fabric8.kubernetes.client.utils.WatcherToggle.eventReceived(WatcherToggle.java:49)
        at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$1.onMessage(WatchConnectionManager.java:237)
        at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:323)
        at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
        at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
        at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:274)
        at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:214)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
2022-01-18 09:02:30,254[aceSharedPool-1]  [ERROR] [o.e.c.c.t.TracingInterceptor 63]     - Error occured while waiting for Pod creation. -id: workspacez53xd6hwh98217ok.ws -message: Websocket connection closed before Pod creation event received
org.eclipse.che.api.workspace.server.spi.InfrastructureException: Error occured while waiting for Pod creation. -id: workspacez53xd6hwh98217ok.ws -message: Websocket connection closed before Pod creation event received
        at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesDeployments.createDeployment(KubernetesDeployments.java:219)
        at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesDeployments.deploy(KubernetesDeployments.java:172)
        at org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInternalRuntime.doStartMachine(KubernetesInternalRuntime.java:742)
        at org.eclipse.che.core.tracing.TracingInterceptor.invoke(TracingInterceptor.java:61)
        at org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInternalRuntime.startMachines(KubernetesInternalRuntime.java:626)
        at org.eclipse.che.core.tracing.TracingInterceptor.invoke(TracingInterceptor.java:61)
        at org.eclipse.che.workspace.infrastructure.kubernetes.KubernetesInternalRuntime.internalStart(KubernetesInternalRuntime.java:204)
        at org.eclipse.che.api.workspace.server.spi.InternalRuntime.start(InternalRuntime.java:141)
        at org.eclipse.che.core.tracing.TracingInterceptor.invoke(TracingInterceptor.java:61)
        at org.eclipse.che.api.workspace.server.WorkspaceRuntimes$StartRuntimeTask.run(WorkspaceRuntimes.java:960)
        at org.eclipse.che.commons.lang.concurrent.CopyThreadLocalRunnable.run(CopyThreadLocalRunnable.java:38)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(Unknown Source)
        at io.opentracing.contrib.concurrent.TracedRunnable.run(TracedRunnable.java:30)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
```

### Additional context

Installation method chectl 7.18.0
Che 7.18.0
Kubernetes Server Version: v1.15.6-r1-20.3.0.2.B001-15.30.2 
Cloud Orange Business Services

**Comment by sparkoo** [comment]
Che 7.18.0 is very old. I'm afraid, I can only suggest to upgrade to latest version.

The error happened while watching the kubernetes resource so it's using websockets. What you've tested with `./kubectl get pod -n che` is simple GET request. You can try to test it with something like `kubectk logs <podname> -n che`, but I'm not 100% sure it's using websockets either.

I would bet that issue is happening because of old fabric8 Kubernetes client in this old Che version and some incompatibility with Kubernetes, which is also very old.

**Comment by gazarenkov** [comment]
@jegithubpro Would you mind to switch to the newer version? 
There are no chance to have this issue fixed for 7.18
Feel free to reopen if you can see it on the newer.
